### PR TITLE
fix(dev): rename superBlock with v7 to certSlug

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Related to #41563

Renames the `superBlock` variable to `certSlug` in the instances where the value contained `-v7`, as this is solely used for certification routing, and must never be confused with the actual `superBlock`.

### CODE REVIEW

**Comment by ShaunSHamilton** [comment]
@ojeytonwilliams I have scattered some comments throughout with suggestions. The main suggestion being to decide on what to do with:
```
challengeMeta.superBlock === "Scientific Computing with Python"
allMarkdownRemark.nodes.frontmatter.superBlock === "Scientific Computing with Python"
superBlockDashedName === "scientific-computing-with-python"
```
My suggestion is:
- Call anything _un-dashed_ as `superBlockTitle` (see below)
- Anything _dashed_ as just plain old `superBlock`

Also, the `client/src/pages/learn` directory with the `.md` files: This did come up in our meeting, but what is the best course of action here:
- All `<superBlock>/index.md` files have `superBlock` as the dashed name
- All other `<superBlock>/<block>/index.md` files have `superBlock` as the _what I call_ `superBlockTitle` (un-dashed name).

Does my suggestion sound reasonable?

File: client/src/templates/Introduction/SuperBlockIntro.js [hunk 1]
```diff
@@ -143,6 +143,8 @@ class SuperBlockIntroductionPage extends Component {
     const nodesForSuperBlock = edges.map(({ node }) => node);
     const blockDashedNames = uniq(nodesForSuperBlock.map(({ block }) => block));
 
+    // This does not even work, as markdownRemark.frontmatter.superBlock
+    // yields the superBlock Title (non-dashed)???
```
Comment (ojeytonwilliams) [inline client/src/templates/Introduction/SuperBlockIntro.js:147]: We don't use the super block intro markdown any more.  AFAIK, all the intros are in JSON files now.

**Comment by ojeytonwilliams** [comment]
> - Call anything un-dashed as superBlockTitle (see below)
> - Anything dashed as just plain old superBlock

Sounds reasonable.  If we can be consistent across titles, blocks and superBlocks that would be great.  It's a bit long, but  `thing` and `dashedThing` is probably the way to go.

File: api-server/src/server/boot/certificate.js [hunk 2]
```diff
@@ -66,9 +66,11 @@ export function getFallbackFrontEndDate(completedChallenges, completedDate) {
   return latestCertDate ? latestCertDate : completedDate;
 }
 
+const slugs = Object.keys(slugCertTypeMap);
+
 function ifNoSuperBlock404(req, res, next) {
-  const { superBlock } = req.body;
-  if (superBlock && superBlocks.includes(superBlock)) {
+  const { slug } = req.body;
```
Comment (ojeytonwilliams) [inline api-server/src/server/boot/certificate.js:72]: `slug` is a bit generic.  It's a lot easier to maintain code when the name is a bit more specific, like, `certSlug`, `superBlockPath` or whatever is the description of how it's used.  Reason being, `slug` already appears in 30 files, so (unless they're all references to this) there's potential for confusion.

Comment (ShaunSHamilton) [reply api-server/src/server/boot/certificate.js:72]: Is this only used, and do we only ever plan to use this, to refer to the endpoint for the certs?

Comment (ojeytonwilliams) [reply api-server/src/server/boot/certificate.js:72]: Ironically, I'm not 100% sure because `superBlock` is used in so many places.  That said, I'm reasonably confident it only has this one use.

**Review by ahmaxed — DISMISSED** [review]

File: config/certification-settings.js [hunk 3]
```diff
@@ -75,6 +75,27 @@ const superBlockCertTypeMap = {
   'machine-learning-with-python-v7': certTypes.machineLearningPyV7
 };
 
+const superBlockCertTypeMap = {
```
Comment (moT01) [inline config/certification-settings.js:78]: Is this still necessary? I don't see it being used anywhere.

**Review by moT01 — CHANGES_REQUESTED** [review]
The link to a cert on the superBlockIntro pages is breaking things for me:

Apr-13-2021 13-10-35

Changing [this line](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/templates/Introduction/components/CertChallenge.js#L64) to use `cert.certSlug` seems to fix it.

File: config/certification-settings.js (hunk 3 above)
Comment (ShaunSHamilton) [reply config/certification-settings.js:78]: I plan on using this, in #41147. Otherwise, I think I should be using it in this PR... will need to double check.

**Comment by ShaunSHamilton** [comment]
Thanks, for catching that, @moT01 .

File: api-server/src/server/boot/certificate.js [hunk 4]
```diff
@@ -337,7 +337,7 @@ function createShowCert(app) {
   return function showCert(req, res, next) {
     let { username, cert } = req.params;
     username = username.toLowerCase();
-    const certType = superBlockCertTypeMap[cert];
+    const certType = certSlugTypeMap[cert];
```
Comment (ahmadabdolsaheb) [inline api-server/src/server/boot/certificate.js:340]: [cert] should be [certSlug], the relative naming on the certification page and showCertification component, etc should be adjusted accordingly. Although it could be resolved in another pr.

Comment (ahmadabdolsaheb) [inline api-server/src/server/boot/certificate.js:340]: @ShaunSHamilton, [cert] should be [certSlug]. The relative naming on the certification page and showCertification component, etc should be adjusted accordingly. Although it could be resolved in another pr.

Comment (ShaunSHamilton) [reply api-server/src/server/boot/certificate.js]: Thanks, for catching this. I have pushed the changes.

**Review by moT01 — APPROVED** [review]
Well, I looked over this pretty thoroughly last time, and again this time. I couldn't find any issues.

LGTM 🎉

**Review by ahmaxed — APPROVED** [review]
